include/seqan3/alignment/pairwise/detail/policy_optimum_tracker_simd.hpp

```diff
@@ -0,0 +1,201 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::policy_optimum_tracker.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <limits>
+#include <seqan3/std/ranges>
+
+#include <seqan3/alignment/pairwise/detail/policy_optimum_tracker.hpp>
+#include <seqan3/core/simd/simd.hpp>
+#include <seqan3/core/simd/simd_algorithm.hpp>
+#include <seqan3/core/simd/simd_traits.hpp>
+#include <seqan3/core/type_traits/lazy.hpp>
+#include <seqan3/range/views/zip.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief A binary operation to update the alignment optimum for the vectorised global alignment algorithm.
+ * \ingroup pairwise_alignment
+ *
+ * \tparam simd_t The simd index type; must model seqan3::simd::simd_concept.
+ *
+ * \details
+ *
+ * Updates the current alignment optimum with the new score and the respective coordinate only if the coordinate
+ * matches the stored column and row index for the respective alignment. The respective coordinates need to be
+ * pre-computed.
+ */
+struct max_score_updater_simd_global
```

Comment:
It is not about the simd functionality. It is about the structure which is quite different to the scalar version. The entire computation is completely the same. But simd means that we process multiple sequences of possibly different length in a single operation. And that needs some specific structural changes. These changes are reflected by living in a dedicated class, so that any developer can follow what happens in which context.